### ISSUE TITLE
Proposal for updating Open UI Charter

### ISSUE THREAD

**Issue by gregwhitworth** [openui/open-ui#670]
The Open UI CG will turn three on March 27. This group has a lot to be proud of as it has seen numerous research efforts and proposals come through the group from tabs/panelsets, breadcrumbs, checkbox, focusgroup, selectmenu, popover API. Some of these efforts resulted in engagements in other standards bodies such as the CSSWG for anchor position and toggles. The Popover API is officially a standard in the WHATWG HTML specification.

But if I'm honest, the best thing that Open UI has done is cultivate a place that is inviting and has brought seasoned spec editors with decades of experience, web browser implementers, design system and component library developers & designers, accessibility experts into one place to try and provide the web platform with a wonderful UI toolkit!

Over these three years, we have learned a lot; and as such a few of us thought it would be good to update the charter to represent the reality of what the group is working on. On the March 17th telecon, I will present this proposal and the key changes made to the charter and why. We will then discuss adopting it formally as our CG charter.

I welcome everyone to review it and please provide feedback. You all have helped make this group and I want to ensure that this charter reflects our mission, vision and expectations for UI on the web. If you don't have any feedback but have reviewed it and approve of it, please leave a thumbs up or +1 emoji on this post.

Thank you.

-----------------------------------------

Proposed New Charter
--------------------

### Background

Since the beginning, web browsers have provided form controls and other website-level UI controls for web designers and developers to leverage in creating interfaces for their users. The web became what it is today precisely because websites are interactive. People search, select, log in, upload content, purchase, ask for what they need --- all through the power of controls.

Having the building blocks of user interfaces supplied by HTML makes it far easier for web developers to create interfaces for their users. Browser-supplied form and UI controls robustly respond to user needs. The same control can behave differently on different types of devices. They provide deep, accessible integration with all hardware and software. They adjust to unexpected conditions precisely because they are handled by the browser. 

The first set of form controls were specified in the early days of HTML, in [1993](https://www.w3.org/MarkUp/HTMLPlus/htmlplus_1.html). They allow for entering text, pressing buttons, uploading files, checking checkboxes, and selecting radio buttons. From the beginning, the visual design of these form controls has been dictated by the browser, often designed to blend in with the underlying operating system.

Over a decade later, by 2004--2008, it was clear more form and website-level UI controls were needed. "Web 2.0" had energized interactivity on the web, and designers were asking for user interfaces that HTML did not support. To implement these designs, web developers turned to third-party plugins like Flash, non-standard technology like ActiveX, or emerging frameworks like jQuery. As the culture of using the web ignited around the world, incredibly-common interaction design patterns became the new normal.

HTML5 was born of the idea of "[paving the cowpaths](https://www.w3.org/TR/html-design-principles/#pave-the-cowpaths)". The WHAT Working Group (WHATWG) [embarked](https://lists.w3.org/Archives/Public/public-html/2007Apr/0429.html) on an effort to take the most popular patterns, seen on millions of websites, and codify those patterns in HTML itself. Web developers would not need plug-ins or scripts, they could simply use HTML to create rich forms for entering a number, a URL, a password, or an email address; choosing a date and time; filling out a search field; picking a color; choosing a range from a slider; or choosing an item from a predefined list. For the first time, web developers could use HTML for creating placeholder text, specifying which field should capture focus, or making a form field required. The new HTML elements made it incredibly easy to build forms that are faster, more secure, and completely accessible.

But another 15 years have gone by since the last major revision to form and UI interfaces. Most complex web projects today need far more than what HTML5 form and UI controls provide. And so, all-too-often today's web developers turn to heavy JavaScript frameworks to build the user interfaces their designers are asking for. These custom interfaces are rarely fully accessible. They slow down the page, consume power, open security vulnerabilities and exclude people. 

Additionally, today's projects often reject existing built-in form and UI controls because they require more agency over the look and feel of the interface. Web designers demand the ability to express their project's branding across every element on the page, including all user interfaces. Current form and website-level UI controls do not allow for this.

### Vision

It's time to modernize HTML once again, and standardize the underlying technology needed by web developers to create the most common patterns of form and website-level UI controls. HTML alone won't be enough, however. New CSS and JavaScript are needed as well. And rather than simply standardize specific interfaces, it's time to create a more powerful underlying architecture so the creators of web sites and apps can design and build their own ideas for interfaces.

The Open UI Community Group is tasked with facilitating a larger architectural plan for how HTML, CSS, JS, and Web APIs can be combined to provide needed technology so web developers can create modern custom user interfaces. 

Once foundational planning is done, each needed web standard will be defined in the appropriate working group. Open UI is tasked with creating recommendations for those working groups, not defining the standards themselves. 

### Scope of work

The Open UI community group is focused on improving form controls and other website-level UI controls on the web by pursuing the following:

-   **Research**
    -   Document universal component patterns seen in popular 3rd-party web development frameworks.
    -   Capture commonly-used language for component names and parts, states, behaviors, and transition triggers.
    -   Conduct informal developer-facing user research.
    -   Discover gaps or bugs in what's possible with current web technology.
-   **Plan**
    -   Debate and define guiding Open UI Design Principles.
    -   Define developer needs discovered by research.
    -   Document group progress at [open-ui.org](https://open-ui.org/) and [github.com/openui/open-ui](https://github.com/openui/open-ui).
-   **Recommend**
    -   Write draft proposals for targeted improvements to form controls and other website-level UI controls, their styling, and behavior to HTML, CSS, JavaScript, Accessibility or Web APIs.
    -   Deliver the draft proposals as suggestions to the WHATWG, CSSWG, W3C, TC39 and other relevant standards bodies for further debate, adoption, and for implementation in browsers after becoming official HTML, CSS, JS, ARIA or Web API specifications. 
    -   Open issues for any bugs discovered with relevant browsers or working groups.

### Out of scope

-   Any design, innovation, invention or specification of novel or unique UI patterns or controls, or features that are iconic to a specific platform.
-   Any attempt to determine or specify the default look or behavior of these controls in the context of a particular operating system or hardware device.
